xrpl/wallet/main.py

```diff
@@ -15,48 +13,43 @@ class Wallet:
     details.
     """
 
-    def __init__(self: Wallet, seed: str) -> None:
+    def __init__(self: Wallet, public_key: str, private_key: str) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
+            public_key: The public key for this Wallet.
+            private_key: The private key for this Wallet.
         """
-        #: The core value that is used to derive all other information about
-        #: this wallet. MUST be kept secret!
-        self.seed = seed
-
-        pk, sk = derive_keypair(self.seed)
         #: The public key that is used to identify this wallet's signatures, as
         #: a hexadecimal string.
-        self.pub_key = pk
+        self.public_key = public_key
         #: The private key that is used to create signatures, as a hexadecimal
         #: string. MUST be kept secret!
-        self.priv_key = sk
+        self.private_key = private_key
 
         #: The address that publicly identifies this wallet, as a base58 string.
-        self.classic_address = derive_classic_address(self.pub_key)
+        self.classic_address = derive_classic_address(self.public_key)
 
         #: The next available sequence number to use for transactions from this
         #: wallet.
         self.next_sequence_num = 0
 
     @classmethod
-    def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.SECP256K1
-    ) -> Wallet:
+    def create(cls: Type[Wallet], seed: Optional[str] = None) -> Wallet:
```

Comment:
I think you should still be able to pick a crypto algorithm explicitly. There have been a couple cases in the past and are likely to be more in the future where the user might want to choose a specific type. Also, people who are importing secp256k1 keys generated by other software might want to use this class to do that.

Reply:
I still think you should be able to request a non-default crypto algorithm here.

Reply:
yeah i think my chagnes to `Wallet` need to be backed out, with the exception of defaulting to ED25519